.clang-tidy

```diff
@@ -23,14 +23,28 @@ WarningsAsErrors: ''
 HeaderFilterRegex: ''
 UseColor: true
 CheckOptions:
-  - { key: misc-const-correctness.AnalyzeValues,              value: false }
-  - { key: readability-identifier-naming.NamespaceCase,       value: CamelCase }
-  - { key: readability-identifier-naming.ClassCase,           value: CamelCase  }
-  - { key: readability-identifier-naming.StructCase,          value: CamelCase  }
-  - { key: readability-identifier-naming.FunctionCase,        value: camelBack }
-  - { key: readability-identifier-naming.VariableCase,        value: camelBack }
-  - { key: readability-identifier-naming.GlobalConstantCase,  value: UPPER_CASE }
-  - { key: readability-identifier-naming.StaticConstantCase,  value: UPPER_CASE }
-  - { key: readability-identifier-naming.StaticVariableCase,  value: UPPER_CASE }
-  - { key: cppcoreguidelines-avoid-do-while.IgnoreMacros,     value: true }
+  - { key: readability-identifier-naming.NamespaceCase,            value: CamelCase }
+  - { key: readability-identifier-naming.ClassCase,                value: CamelCase  }
+  - { key: readability-identifier-naming.StructCase,               value: CamelCase  }
+  - { key: readability-identifier-naming.TemplateParameterCase,    value: CamelCase  }
+  - { key: readability-identifier-naming.FunctionCase,             value: camelCase  }
+  - { key: readability-identifier-naming.VariableCase,             value: camelCase }
+  - { key: readability-identifier-naming.ClassMemberCase,          value: camelCase }
+  - { key: readability-identifier-naming.ClassMemberSuffix,        value: _          }
+  - { key: readability-identifier-naming.PrivateMemberSuffix,      value: _          }
+  - { key: readability-identifier-naming.ProtectedMemberSuffix,    value: _          }
```

Comment:
We tend to use snake_case for private/internal members (including for members that are "logically" private but exposed publicly for logging/debugging).  Not sure if clang-tidy provides a way to note that.

Reply:
Yes, that is possible with https://clang.llvm.org/extra/clang-tidy/checks/readability/identifier-naming.html#cmdoption-arg-PrivateMemberCase

Reply:
At least in the compiler front end and libraries, the common style seems to be `camelCase` for everything. We can easily change that with an automated `clang-tidy` run.